etc/migration/10.3-to-11.0/mapped-emails.js

```diff
@@ -0,0 +1,180 @@
+/*!
+ * Copyright 2015 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * This migration script will try to create a mapping between a user's email address and the user
+ * object in the Principals table. Before it does that however, the following checks will be made:
+ *   -  Ensure that the user has an email address
+ *   -  Ensure that the user's email address is valid
+ */
+
+var _ = require('underscore');
+var bunyan = require('bunyan');
+var csv = require('csv');
+var fs = require('fs');
+var optimist = require('optimist');
+
+var Cassandra = require('oae-util/lib/cassandra');
+var log = require('oae-logger').logger('email-mapping-migrator');
+var OAE = require('oae-util/lib/oae');
+var PrincipalsDAO = require('oae-principals/lib/internal/dao');
+var Validator = require('oae-util/lib/validator').Validator;
+
+
+var argv = optimist.usage('$0 [--config <path/to/config.js>] [--warnings <path/to/warnings.csv>]')
+    .alias('c', 'config')
+    .describe('c', 'Specify an alternate config file')
+    .default('c', __dirname + '/config.js')
+
+    .alias('w', 'warnings')
+    .describe('w', 'Specify the path to the file where unmappable users should be dumped to')
+    .default('w', __dirname + '/email-migration.csv')
+
+    .alias('h', 'help')
+    .describe('h', 'Show usage information')
+    .argv;
+
+if (argv.help) {
+    optimist.showHelp();
+    return;
+}
+
+var warningFilePath = argv.warnings;
+
+// If a relative path that starts with `./` has been provided,
+// we turn it into an absolute path based on the current working directory
+if (argv.config.match(/^\.\//)) {
+    argv.config = process.cwd() + argv.config.substring(1);
+// If a different non-absolute path has been provided, we turn
+// it into an absolute path based on the current working directory
+} else if (!argv.config.match(/^\//)) {
+    argv.config = process.cwd() + '/' + argv.config;
+}
```

Comment:
I think you can just do `path.resolve(process.cwd(), argv.config);` to replace all of this. It should do both the relative and absolute `argv.config` path handling for you.